lib/c-api/src/wasm_c_api/wasi/mod.rs

```diff
@@ -304,7 +305,124 @@ pub unsafe extern "C" fn wasi_get_wasi_version(module: &wasm_module_t) -> wasi_v
         .unwrap_or(wasi_version_t::InvalidVersion)
 }
 
-/// Takes ownership of `wasi_env_t`.
+/// Non-standard type wrapping `wasm_extern_t` with the addition of
+/// two `wasm_name_t` respectively for the module name and the name of
+/// the extern (very likely to be an import). This non-standard type
+/// is used by the non-standard `wasi_get_unordered_imports` function.
+#[allow(non_camel_case_types)]
+#[derive(Clone)]
+pub struct wasm_named_extern_t {
+    module: Box<wasm_name_t>,
+    name: Box<wasm_name_t>,
+    r#extern: Box<wasm_extern_t>,
+}
+
+wasm_declare_boxed_vec!(named_extern);
+
+/// Non-standard function to get the module name of a
+/// `wasm_named_extern_t`.
+#[no_mangle]
+pub extern "C" fn wasm_named_extern_module(named_extern: &wasm_named_extern_t) -> &wasm_name_t {
+    named_extern.module.as_ref()
+}
+
+/// Non-standard function to get the name of a `wasm_named_extern_t`.
+#[no_mangle]
+pub extern "C" fn wasm_named_extern_name(named_extern: &wasm_named_extern_t) -> &wasm_name_t {
+    named_extern.name.as_ref()
+}
+
+/// Non-standard function to get the wrapped extern of a
+/// `wasm_named_extern_t`.
+#[no_mangle]
+pub extern "C" fn wasm_named_extern_extern(named_extern: &wasm_named_extern_t) -> &wasm_extern_t {
```

Comment:
Review use of `extern` here, it seems the use of `extern` in the Wasm C API is different than that of the Wasm spec / our API.

Reply:
What I mean by that is, let's consider possibly renaming this. haha

Reply:
Yup, totally agree.

My proposal:

1. `wasm_named_extern_inner`,
2. `wasm_named_extern_unwrap`.

What's your favorite?

Reply:
(I've renamed the function `wasm_named_extern_unwrap` for the moment).